### ISSUE TITLE
doom/find-file-in-private-config doesn't work if .doom is a symlink

### ISSUE THREAD

**Issue by hhamud** [doomemacs/doomemacs#7312]
### I confirm that...

- [X] I have searched the [issue tracker](https://github.com/doomemacs/doomemacs/issues), [documentation](https://docs.doomemacs.org), [FAQ](https://docs.doomemacs.org/-/faq), [Discourse](https://discourse.doomemacs.org), and [Google](https://google.com), in case this issue has already been reported/resolved.

- [X] I have read "[How to Debug Issues](https://doomemacs.org/d/how2debug)", and will use it to provide as much information about this issue as possible.

- [X] The issue can be reproduced on the **latest** available commit of Doom.

- [X] The issue can be reproduced on a stable release of Emacs, such as 27 or 28. *(Doom does not support development builds like 29+ or any version ending in .50 or .9x)*

### Expected behavior

When using SPC-f-p, I expected it to find my private config files but it failed to do so.

### Current behavior

I am having an issue with doom/find-file-in-private-config command where .doom.d dir is symlinked to my dotfiles  

Here is the full error:

```elisp
Debugger entered--Lisp error: (cl-no-applicable-method project-root "/Users/user/.doom.d/")
  signal(cl-no-applicable-method (project-root "/Users/user/.doom.d/"))
  cl-no-applicable-method(#s(cl--generic :name project-root :dispatches (((&context . project--within-roots-fallback) #s(cl--generic-generalizer :name cl--generic-eql-generalizer :priority 100 :tagcode-function #f(compiled-function (name &rest _) #<bytecode -0x1cba097e092fa1fc>) :specializers-function #f(compiled-function (tag &rest _) #<bytecode -0xc2bce8c284cdf9a>)) #s(cl--generic-generalizer :name cl--generic-t-generalizer :priority 0 :tagcode-function #f(compiled-function (name &rest _) #<bytecode -0x1cba0b13a96764e4>) :specializers-function #f(compiled-function (tag &rest _) #<bytecode -0x2b662d08469105e>))) (0 #s(cl--generic-generalizer :name cl--generic-head-generalizer :priority 80 :tagcode-function #f(compiled-function (name &rest _) #<bytecode 0x3a10684aa2727b5>) :specializers-function #f(compiled-function (tag &rest _) #<bytecode -0xc2bceedda9ba79a>)) #s(cl--generic-generalizer :name cl--generic-t-generalizer :priority 0 :tagcode-function #f(compiled-function (name &rest _) #<bytecode -0x1cba0b13a96764e4>) :specializers-function #f(compiled-function (tag &rest _) #<bytecode -0x2b662d08469105e>)))) :method-table (#s(cl--generic-method :specializers ((head projectile)) :qualifiers nil :call-con nil :function #f(compiled-function (project) #<bytecode 0x6b9aca8ac27aec5>)) #s(cl--generic-method :specializers ((head vc)) :qualifiers nil :call-con nil :function #f(compiled-function (project) #<bytecode 0x6b9aca9b752fa79>)) #s(cl--generic-method :specializers ((head transient)) :qualifiers nil :call-con nil :function #f(compiled-function (project) #<bytecode 0x6b9aca8ac27aec5>)) #s(cl--generic-method :specializers (t ((&context . project--within-roots-fallback) eql nil)) :qualifiers nil :call-con nil :function #f(compiled-function (project) #<bytecode 0x6a78faed84a2479>))) :options nil) "/Users/user/.doom.d/")
  apply(cl-no-applicable-method #s(cl--generic :name project-root :dispatches (((&context . project--within-roots-fallback) #s(cl--generic-generalizer :name cl--generic-eql-generalizer :priority 100 :tagcode-function #f(compiled-function (name &rest _) #<bytecode -0x1cba097e092fa1fc>) :specializers-function #f(compiled-function (tag &rest _) #<bytecode -0xc2bce8c284cdf9a>)) #s(cl--generic-generalizer :name cl--generic-t-generalizer :priority 0 :tagcode-function #f(compiled-function (name &rest _) #<bytecode -0x1cba0b13a96764e4>) :specializers-function #f(compiled-function (tag &rest _) #<bytecode -0x2b662d08469105e>))) (0 #s(cl--generic-generalizer :name cl--generic-head-generalizer :priority 80 :tagcode-function #f(compiled-function (name &rest _) #<bytecode 0x3a10684aa2727b5>) :specializers-function #f(compiled-function (tag &rest _) #<bytecode -0xc2bceedda9ba79a>)) #s(cl--generic-generalizer :name cl--generic-t-generalizer :priority 0 :tagcode-function #f(compiled-function (name &rest _) #<bytecode -0x1cba0b13a96764e4>) :specializers-function #f(compiled-function (tag &rest _) #<bytecode -0x2b662d08469105e>)))) :method-table (#s(cl--generic-method :specializers ((head projectile)) :qualifiers nil :call-con nil :function #f(compiled-function (project) #<bytecode 0x6b9aca8ac27aec5>)) #s(cl--generic-method :specializers ((head vc)) :qualifiers nil :call-con nil :function #f(compiled-function (project) #<bytecode 0x6b9aca9b752fa79>)) #s(cl--generic-method :specializers ((head transient)) :qualifiers nil :call-con nil :function #f(compiled-function (project) #<bytecode 0x6b9aca8ac27aec5>)) #s(cl--generic-method :specializers (t ((&context . project--within-roots-fallback) eql nil)) :qualifiers nil :call-con nil :function #f(compiled-function (project) #<bytecode 0x6a78faed84a2479>))) :options nil) "/Users/user/.doom.d/")
  #f(compiled-function (&rest args) #<bytecode 0xfb02fee7f108dc>)("/Users/user/.doom.d/")
  apply(#f(compiled-function (&rest args) #<bytecode 0xfb02fee7f108dc>) "/Users/user/.doom.d/" nil)
  #f(compiled-function (arg &rest args) #<bytecode -0x97003a6edef9106>)("/Users/user/.doom.d/")
  apply(#f(compiled-function (arg &rest args) #<bytecode -0x97003a6edef9106>) "/Users/user/.doom.d/")
  project-root("/Users/user/.doom.d/")
  project-roots("/Users/user/.doom.d/")
  #f(compiled-function (project) #<bytecode 0x6a78faed84a2479>)("/Users/user/.doom.d/")
  apply(#f(compiled-function (project) #<bytecode 0x6a78faed84a2479>) "/Users/user/.doom.d/" nil)
  #f(compiled-function (arg &rest args) #<bytecode -0xb6403c988392103>)("/Users/user/.doom.d/")
  apply(#f(compiled-function (arg &rest args) #<bytecode -0xb6403c988392103>) "/Users/user/.doom.d/")
  project-root("/Users/user/.doom.d/")
  #f(compiled-function (project &optional dirs) #<bytecode -0x6116af23c3fa3bd>)("/Users/user/.doom.d/" nil)
  apply(#f(compiled-function (project &optional dirs) #<bytecode -0x6116af23c3fa3bd>) "/Users/user/.doom.d/" nil)
  project-files("/Users/user/.doom.d/" nil)
  project-find-file-in(nil nil "/Users/user/.doom.d/")
  (cond ((and projectile-project-root (file-equal-p projectile-project-root default-directory)) (if (doom-project-p default-directory) nil (setq projectile-enable-caching nil)) (call-interactively (if (doom-module-p :completion 'ivy) #'counsel-projectile-find-file #'projectile-find-file))) ((and (and (boundp 'vertico-mode) vertico-mode) (fboundp '+vertico/find-file-in)) (+vertico/find-file-in default-directory)) ((and (and (boundp 'ivy-mode) ivy-mode) (fboundp 'counsel-file-jump)) (call-interactively #'counsel-file-jump)) ((project-current nil dir) (project-find-file-in nil nil dir)) ((and (and (boundp 'helm-mode) helm-mode) (fboundp 'helm-find-files)) (call-interactively #'helm-find-files)) ((call-interactively #'find-file)))
  (let* ((default-directory (file-truename dir)) (projectile-project-root (doom-project-root dir)) (projectile-enable-caching projectile-enable-caching)) (cond ((and projectile-project-root (file-equal-p projectile-project-root default-directory)) (if (doom-project-p default-directory) nil (setq projectile-enable-caching nil)) (call-interactively (if (doom-module-p :completion 'ivy) #'counsel-projectile-find-file #'projectile-find-file))) ((and (and (boundp 'vertico-mode) vertico-mode) (fboundp '+vertico/find-file-in)) (+vertico/find-file-in default-directory)) ((and (and (boundp 'ivy-mode) ivy-mode) (fboundp 'counsel-file-jump)) (call-interactively #'counsel-file-jump)) ((project-current nil dir) (project-find-file-in nil nil dir)) ((and (and (boundp 'helm-mode) helm-mode) (fboundp 'helm-find-files)) (call-interactively #'helm-find-files)) ((call-interactively #'find-file))))
  doom-project-find-file("/Users/user/.doom.d/")
  doom/find-file-in-private-config()
  funcall-interactively(doom/find-file-in-private-config)
  command-execute(doom/find-file-in-private-config)

```

Here I have checked if it is symlinked:
```shell
user@Users-Laptop ~ % ls -al .doom.d 
lrwxr-xr-x  1 user  staff  29 30 Dec  2022 .doom.d -> /Users/user/.dotfiles/.doom.d
```

### Steps to reproduce

1. Open Emacs
2. use keyboard shortcut SPC-f-p
3. attempt to select the private configuration files

### System Information

https://pastebin.com/7FuJM9by

**Comment by johnabs** [comment]
I'm having a similar issue, but it seems to be an issue with consult based on the error message (my doom.d is NOT symlinked though).

Basically when I run the function, I simply get: Wrong type argument: overlayp, nil

But other doom functions like Spc f r, Spc RET, etc. all work.

doom doctor is clean, and everything was recently upgraded to emacs 29.1

To clarify, does yours allow you to display the overlay with the files, but not open the files, or does yours just not even open the consult overlay to display the files (I'm experiencing the latter, but I can't tell from your post which one you're experiencing).

Also, did you try removing/renaming the symlink temporarily and moving the doom.d folder to your home directory? If so, does it work then?

**Comment by yonghuier** [comment]
Hi, I have the same issue. When I enter SPC-f-p, I get "!/0 Fd (~/.doom.d): #", but  what I got before was a list of files.

**Comment by hlissner** [comment]
As of 15339e4, the original error should be fixed. @yonghuier's issue, 

> Hi, I have the same issue. When I enter SPC-f-p, I get "!/0 Fd (~/.doom.d): #", but what I got before was a list of files.

@yonghuier Press `SPC-u SPC-f-p`. The leading `SPC-u` forces projectile to clear the file cache, which is likely your issue (and is unrelated to the original error reported in this issue).

**Comment by yonghuier** [comment]
Hi, @hlissner, I press `SPC-u SPC-f-p`, but it doesn't work.  After I restarted Emacs, it still is `!/0 Fd (~/.doom.d): #`.

**Comment by hlissner** [comment]
@yonghuier I see the issue. `doom-project-find-file` was changed in 60e22fd2eb61 to use `consult-fd` for Vertico users, which presents an empty file list at first (it's waiting for you to enter 2+ characters before it starts matching and showing results). This isn't a bug, just misunderstood behavior; but in any case, it is an unintended change, so I've pushed a fix in 1b0af3bfc747d9ef4973af0b5698b1c71cff86f5. In any case, it is not related to this issue. Thanks for letting me know, though!
